garden-backend-service/src/api/dependencies/auth.py

```diff
@@ -0,0 +1,34 @@
+import logging
+from fastapi import Depends, HTTPException
+from fastapi.security import HTTPAuthorizationCredentials, HTTPBearer
+
+from src.auth.auth_state import AuthenticationState
+
+
+logger = logging.getLogger("app")
+
+
+def _get_auth_token(
+    authorization: HTTPAuthorizationCredentials = Depends(HTTPBearer(auto_error=False)),
```

Comment:
Learning about FastAPI dependency injection from this PR, seems very nice to work with